docs/gsoc2020_ideas.rst

```diff
@@ -0,0 +1,134 @@
+3. Look at the guidelines on how to contribute to Scipy.
+4. Make a enhancement/bugfix/documentation fix -- it does not have to be big,
+   and it does not need to be related to your proposal. Doing so before
+   applying for the GSoC is a hard requirement for ``uarray``. It helps
+   everyone you get some idea how things would work during GSoC.
+5. Start writing your proposal early, post a draft to the the mentors' email
+   addresses mailing list and iterate based on the feedback you receive. This
+   will both improve the quality of your proposal and help you find a suitable mentor.
+
+Contact
+-------
+
+If you have a question *after checking all guideline pages above*, you can
+email the mentors.
+
+``uarray`` project ideas
+------------------------
+
+``uarray``: Add querying for state
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Adding querying for the ``uarray._BackendState`` object will allow users of
+``uarray`` to see what's inside the opaque object. Some parts can be re-used
+from the pickling machinery.
+
+It can also help downstream users to access the parameters of the currently
+set backend, which is a planned feature of ``uarray``. Here is a list of goals
+for this project:
+
+* Allow downstream projects to query the list of backends.
+* Allow downstream projects to query the list of parameters for a backend.
+
+This would enable, for example, the following use-cases:
+
+* Allow a downstream library to detect a backend and run specialised code for
+  it.
+* Allow a downstream library to fail-fast on a known-unsupported backend.
+
+This project has a straightforward design and needs some implementation work,
+and will require interacting with the mentors to implement and polish.
+
+* Required knowledge: Python C-API and C++
+* Difficulty level: easy
+* Potential mentors: Peter Bell and Hameer Abbasi
+
+``uarray``: Allow subdomains
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This idea would allow a backend to encompass functions from more than one domain.
+
+The primary goal of this project would be:
+
+* Develop a system that allows, via some kind of matching mechanism, to select
+  which domains it supports, **while maintaining backward compatibility**.
+
+This would allow a backend targeting NumPy to also target, for example, the
+``numpy.random`` submodule.
+
+This project has a somewhat complicated design and needs some involved
+implementation work, and will require interacting with the mentors to flesh
+out and work through.
+
+* Required knowledge: Python C-API and C++
+* Difficulty level: easy
+* Potential mentors: Peter Bell and Hameer Abbasi
+
+``unumpy``: Expand overall coverage
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This project is split into two parts:
+
+* Adding further converage of the NumPy API.
+* Adding more backends to NumPy.
+
+We realise this is a large (possibly open-ended) undertaking, and so there
+will need to be a minimum amount of work done in order to pass.
```

Comment:
This isn't a statement that a student can do much with. I suggest to instead say something like "~150 functions left to implement, if time left we would like to see a new backend for Ray implemented".